controllers/infrastructure/byocluster_controller_test.go

```diff
@@ -0,0 +1,137 @@
+		WaitForObjectsToBePopulatedInCache(byoCluster)
+
+		_, err := byoClusterReconciler.Reconcile(ctx, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      byoCluster.Name,
+				Namespace: byoCluster.Namespace}})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should be able to delete ByoCluster", func() {
+		cluster = builder.Cluster(defaultNamespace, "byocluster-deleted").
+			Build()
+		Expect(k8sClientUncached.Create(ctx, cluster)).Should(Succeed())
+		WaitForObjectsToBePopulatedInCache(cluster)
+
+		byoCluster = builder.ByoCluster(defaultNamespace, "byocluster-deleted").
+			WithOwnerCluster(cluster).
+			Build()
+		Expect(k8sClientUncached.Create(ctx, byoCluster)).Should(Succeed())
+		WaitForObjectsToBePopulatedInCache(byoCluster)
+
+		byoClusterLookupKey := types.NamespacedName{Name: byoCluster.Name, Namespace: byoCluster.Namespace}
+		_, err := byoClusterReconciler.Reconcile(ctx, reconcile.Request{
+			NamespacedName: byoClusterLookupKey})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(k8sClientUncached.Delete(ctx, byoCluster)).Should(Succeed())
+		WaitForObjectToBeUpdatedInCache(byoCluster, func(object client.Object) bool {
+			return !object.(*infrastructurev1beta1.ByoCluster).ObjectMeta.DeletionTimestamp.IsZero()
+		})
+
+		_, err = byoClusterReconciler.Reconcile(ctx, reconcile.Request{
+			NamespacedName: byoClusterLookupKey})
+		Expect(err).NotTo(HaveOccurred())
+
+		// assert ByoCluster does not exists
+		deletedByoCluster := &infrastructurev1beta1.ByoCluster{}
+		err = k8sClientUncached.Get(ctx, byoClusterLookupKey, deletedByoCluster)
+		Expect(err).To(MatchError(fmt.Sprintf("byoclusters.infrastructure.cluster.x-k8s.io %q not found", byoClusterLookupKey.Name)))
+
+	})
+
+	It("should get valid value of fields when ByoClusterController gets a create request", func() {
+		cluster = builder.Cluster(defaultNamespace, "byocluster-finalizer").
+			Build()
+		Expect(k8sClientUncached.Create(ctx, cluster)).Should(Succeed())
+		WaitForObjectsToBePopulatedInCache(cluster)
+
+		byoCluster = builder.ByoCluster(defaultNamespace, "byocluster-finalizer").
+			WithOwnerCluster(cluster).
+			Build()
+		Expect(k8sClientUncached.Create(ctx, byoCluster)).Should(Succeed())
+		WaitForObjectsToBePopulatedInCache(byoCluster)
+
+		byoClusterLookupKey := types.NamespacedName{Name: byoCluster.Name, Namespace: byoCluster.Namespace}
+		_, err := byoClusterReconciler.Reconcile(ctx, reconcile.Request{
+			NamespacedName: byoClusterLookupKey})
+		Expect(err).NotTo(HaveOccurred())
+
+		createdByoCluster := &infrastructurev1beta1.ByoCluster{}
+		err = k8sClientUncached.Get(ctx, byoClusterLookupKey, createdByoCluster)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(controllerutil.ContainsFinalizer(createdByoCluster, infrastructurev1beta1.ClusterFinalizer)).To(BeTrue())
+		Expect(createdByoCluster.Status.Ready).To(BeTrue())
+		Expect(createdByoCluster.Spec.ControlPlaneEndpoint.Port).NotTo(Equal(0))
```

Comment:
We should check on the default value
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/4e84b4f263e2737fd3f6b71a962b25aa836ee94b/controllers/infrastructure/byocluster_controller.go#L32

Reply:
ok